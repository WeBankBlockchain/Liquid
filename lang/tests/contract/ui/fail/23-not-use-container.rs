use liquid_lang as liquid;

#[liquid::contract]
mod noop {
    #[liquid(storage)]
    struct Noop {
        b: bool,
    }

    #[liquid(methods)]
    impl Noop {
        pub fn new(&mut self) {}

        pub fn noop(&self) {}
    }
}

fn main() {}
