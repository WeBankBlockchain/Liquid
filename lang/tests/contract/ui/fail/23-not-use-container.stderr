error[E0277]: the trait bound `bool: You_Should_Use_A_Container_To_Wrap_Your_State_Field_In_Storage` is not satisfied
 --> $DIR/23-not-use-container.rs:7:12
  |
7 |         b: bool,
  |            ^^^^ the trait `You_Should_Use_A_Container_To_Wrap_Your_State_Field_In_Storage` is not implemented for `bool`
